### ISSUE TITLE
Could GRPOTrainer support pre-generated completions

### ISSUE THREAD

**Issue by evehsu** [huggingface/trl#3663]
### Feature request

GRPOTrainer generate completions given a prompt from the train_dataset; It doesn't use completion even if train_dataset have completion field. Could we add a flag/param to GRPOTrainer to indicate whether to use completions on-the-fly, or use pre generated completions from the train_dataset?

### Motivation

We want to use completion generated from a more advanced LLM to fine-tune a small open sourced LLM. Generating completion on the fly limits us to generate completion only from current small LLM, which has much lower change to generate a high quality (high reward score) completion. 

### Your contribution

TBD

**Comment by qgallouedec** [comment]
Why not using sft instead?

**Comment by LeonEricsson** [comment]
Closing due to inactivity. Feel free to reopen if you'd like to discuss further!
